sympy/assumptions/sathandlers.py

```diff
@@ -190,39 +191,15 @@ def _old_assump_replacer(obj):
     if not isinstance(obj, AppliedPredicate):
         return obj
 
-    e = obj.args[0]
+    e = obj.arguments[0]
     ret = None
 
-    if obj.func == Q.positive:
-        ret = e.is_positive
-    elif obj.func == Q.zero:
-        ret = e.is_zero
-    elif obj.func == Q.negative:
-        ret = e.is_negative
-    elif obj.func == Q.nonpositive:
-        ret = e.is_nonpositive
-    elif obj.func == Q.nonzero:
-        ret = e.is_nonzero
-    elif obj.func == Q.nonnegative:
-        ret = e.is_nonnegative
-
-    elif obj.func == Q.rational:
-        ret = e.is_rational
-    elif obj.func == Q.irrational:
-        ret = e.is_irrational
-
-    elif obj.func == Q.even:
-        ret = e.is_even
-    elif obj.func == Q.odd:
-        ret = e.is_odd
-    elif obj.func == Q.integer:
-        ret = e.is_integer
-    elif obj.func == Q.composite:
-        ret = e.is_composite
-    elif obj.func == Q.imaginary:
-        ret = e.is_imaginary
-    elif obj.func == Q.commutative:
-        ret = e.is_commutative
+    for key in _assume_defined:
+        pred = getattr(Q, key, None)
+        if pred is None:
+            continue
+        if obj.func == pred:
+            ret = getattr(e, as_property(key))
```

Comment:
Rather than a loop I would guess that it would be more efficient to use a dict like:
```
getters = {
    Q.positive: lambda o: o.is_positive,
    ...
}

def replacer():
    getter = getters[obj.func]
```
Note this should not really use `obj.func` but `obj.predicate` or some other property. This class has messed up `func/args` that should be fixed later.

Reply:
How about this?
```python
def _old_assump_replacer(obj):
    if not isinstance(obj, AppliedPredicate):
        return obj

    pred = obj.function
    arg = obj.arguments[0]
    ret = getattr(arg, as_property(pred.name), None)

    if ret is None:
        return obj

    return ret
```

This way the predicate is directly converted to the property without comparing it to every other predicates.

Reply:
I suggest running some timings.

Reply:
Your suggestion was the fastest. I applied that approach.